crates/circuit/src/circuit_instruction.rs

```diff
@@ -695,6 +701,37 @@ impl<'py> FromPyObject<'py> for OperationFromPython {
                 extra_attrs: extract_extra()?,
             });
         }
+        'standard_instr: {
+            // Our Python standard instructions have a `_standard_instruction_type` field at the
+            // class level so we can quickly identify them here without an `isinstance` check.
+            // Once we know the type, we query the object for any type-specific fields we need to
+            // read (e.g. a Barrier's number of qubits) to build the Rust representation.
+            let Some(standard_type) = ob_type
+                .getattr(intern!(py, "_standard_instruction_type"))
```

Comment:
I can change this to `_standard_instruction`, if you prefer. I did this because there's also a Rust-side `StandardInstruction` enum, and even though it isn't exposed to Python, my thinking was that this would be more consistent to readers of the Qiskit codebase if the name here aligned to the `StandardInstructionType` Rust enum that backs it.

Reply:
I can only speak to my interpretation, but to me, the mismatched attributed names jumped out as odd.  I get that it's not super pleasant with needing the `StandardInstruction`/`StandardInstructionType` split anyway, and I don't feel strongly about which way you want to keep it.